### PR TITLE
[L1T] Phase-2, include missing collections in outputs

### PR DESCRIPTION
**Pull request by cecilecaillol**
#### PR description:

This PR addresses the missing Tk object collections in the CMSSW_12_5_0_pre5 RelVals. It includes fixes were needed to avoid errors after adding in the objects to the EventContent. One error that came up in the FEVTDEBUGHLToutput_step was: The class "l1t::TkJetWord" is compiled and for its data member "tkJetWord_" we do not have a dictionary for the collection "bitset<70>". Because of this, we will not be able to read or write this data member.
classes_def.xml was adjusted to avoid the error. Local PR: https://github.com/cms-l1t-offline/cmssw/pull/1048/

This PR also adds missing collections for Phase 2 tau-jet-met to the event content. Specifically the 9x9 Sums for histogrammed jets, the NN Puppi taus, and the HPS taus. Local PR: https://github.com/cms-l1t-offline/cmssw/pull/1047

We need this PR to be backported to 12_5 for our Phase-2 L1 MC production

### CODE REVIEW

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
please test

**Comment by srimanob** [comment]
May I ask for clarification, @cecilecaillol  ?

The added collections will be in FEVTDEBUG only, or also RAWSIM (i.e. future production workflows)?

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by cecilecaillol** [comment]
urgent
(needed for L1 Phase2 MC production in 12_5)

**Comment by cecilecaillol** [comment]
> May I ask for clarification, @cecilecaillol ?
> 
> The added collections will be in FEVTDEBUG only, or also RAWSIM (i.e. future production workflows)?

@srimanob We would like the new collections to be in both.

**Comment by BenjaminRS** [comment]
We have found an issue when testing this PR that needs to be fixed. Most likely the TkJet emulator needs to be updated in order to fix an error message of type:
`L1TrkPtrs Not Assigned!`
This could then be what is causing some other errors further down the chain. For example in `L1TkHTMissEmulatorProducer` we get the error:
`Warning: TkJetCollection not found in the event. Exit`

File: L1Trigger/Configuration/python/L1Trigger_EventContent_cff.py [hunk 1]
```diff
@@ -203,6 +216,8 @@ def _appendPhase2Digis(obj):
         'keep *_l1tMETPFProducer_*_*',
         'keep *_l1tNNTauProducer_*_*',
         'keep *_l1tNNTauProducerPuppi_*_*',
+        'keep *_HPSPFTauProducerPF_*_*',
+        'keep *_HPSPFTauProducerPuppi_*_*',
```
Comment (missirol) [inline L1Trigger/Configuration/python/L1Trigger_EventContent_cff.py:220]: Shouldn't this be `l1tHPS..`?

**Comment by srimanob** [comment]
Hi @BenjaminRS @cecilecaillol 

Seeing that you fix `L1VertexFinderEmulator` in
https://github.com/cms-sw/cmssw/blob/master/L1Trigger/Configuration/python/SimL1Emulator_cff.py#L170

How about it in
https://github.com/cms-sw/cmssw/blob/master/L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc#L233
https://github.com/cms-sw/cmssw/blob/master/L1Trigger/L1TTrackMatch/plugins/L1TrackSelectionProducer.cc#L605
? Do you have plan to update, or later PR? 
I will need the fix for https://github.com/cms-sw/cmssw/pull/39450 

Thanks.

**Comment by BenjaminRS** [comment]
> We have found an issue when testing this PR that needs to be fixed. Most likely the TkJet emulator needs to be updated in order to fix an error message of type:
> `L1TrkPtrs Not Assigned!`
> This could then be what is causing some other errors further down the chain. For example in `L1TkHTMissEmulatorProducer` we get the error:
> `Warning: TkJetCollection not found in the event. Exit`

PR #39479 fixes the issue we saw before and therefore need that PR merged for the MC campaign.

**Comment by BenjaminRS** [comment]
@srimanob I am not sure I understand [your comment](https://github.com/cms-sw/cmssw/pull/39401#issuecomment-1254103709) above. [SimL1Emulator_cff.py](https://github.com/cms-sw/cmssw/blob/master/L1Trigger/Configuration/python/SimL1Emulator_cff.py#L170), [HPSPFTauProducer.cc](https://github.com/cms-sw/cmssw/blob/master/L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc#L233), and [L1TrackSelectionProducer.cc](https://github.com/cms-sw/cmssw/blob/master/L1Trigger/L1TTrackMatch/plugins/L1TrackSelectionProducer.cc#L605) all use the same InputTag in order to call `L1VertexFinderEmulator`.

**Comment by srimanob** [comment]
Hi @BenjaminRS 
I assume you change from `("L1VertexFinder", "l1vertices")` to `("l1tVertexFinder", "l1vertices")`

or I am wrong?

File: L1Trigger/Configuration/python/L1Trigger_EventContent_cff.py (hunk 1 above)
Comment (srimanob) [reply L1Trigger/Configuration/python/L1Trigger_EventContent_cff.py:220]: @cecilecaillol 
Could you please confirm?

**Comment by srimanob** [comment]
@cmsbuild please test

Retrigger the test as it was almost 2 weeks ago.
@cecilecaillol could you please provide information as asked by @missirol in https://github.com/cms-sw/cmssw/pull/39401#discussion_r976148112

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by cecilecaillol** [comment]
> @cmsbuild please test
> 
> Retrigger the test as it was almost 2 weeks ago. @cecilecaillol could you please provide information as asked by @missirol in [#39401 (comment)](https://github.com/cms-sw/cmssw/pull/39401#discussion_r976148112)

Addressed in last commit

**Comment by srimanob** [comment]
+upgrade

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
@cecilecaillol @BenjaminRS please add to the PR description an estimate of the increase of the event size, as I already asked for the backport PR in https://github.com/cms-sw/cmssw/pull/39451#issuecomment-1266526299

**Pull request merged**